src/java.net.http/share/classes/jdk/internal/net/http/Http2ClientImpl.java

```diff
@@ -188,9 +188,7 @@ void deleteConnection(Http2Connection c) {
         if (debug.on())
             debug.log("removing from the connection pool: %s", c);
         synchronized (this) {
```

Comment:
BTW, this `synchronized` seems redundant now too.

Reply:
Yes, possibly. We'd be relying on internal CHM synchronization if we removed that - there may be some side effects with other synchronized blocks in Http2ClientImpl. Let's not go into this right now.